examples/bq_file_load_benchmark/README.md

```diff
@@ -0,0 +1,466 @@
+column types, number of files or file sizes affect the performance of load jobs.
+While Google has high-level guidelines, we don't have hard data on how these 
+parameters specifically affect performance. This project contains the tools to 
+provide the data to answer these questions. 
+
+## File Parameters
+Specifc file parameters are used in this project for performance testing. While 
+the list of parameters is growing, the current list of paramters and values
+is as follows:
+
+##### File Type: 
+
+* Avro
+* CSV
+* JSON
+* Parquet
+
+##### Compression: 
+
+* gzip (for CSV and JSON)
+* snappy (for AVRO)
+
+##### Number of columns
+* 10
+* 100
+* 1000
+
+##### Column Types
+* String-only
+* 50% String / 50% NUMERIC
+* 10% String / 90% NUMERIC
+
+##### Number of files
+* 1
+* 100
+* 1000
+* 10000
+
+##### Target Data Size (Size of the BigQuery staging table used to generate each file)
+* 10MB
+* 100MB
+* 1GB
+* 2GB
+
+These parameters are used to create combinations of file types stored on in a 
+bucket on GCS. An example of a file prefix generated from the above list of file parameters is: 
+fileType=csv/compression=none/numColumns=10/columnTypes=100_STRING/numFiles=100/tableSize=10MB/* .
+This prefix holds 100 uncompressed CSV files, each generated from a 10 MB BigQuery
+staging table with 10 string columns. The tool loads the 100 CSV files with this prefix 
+to a BigQuery table and records the performance to create a benchmark. 
+
+## Benchmark Results
+
+#### BigQuery
+
+The results of the benchmark loads are saved in a seperate BigQuery for 
+ad hoc analysis. The results table contains the following schema: 
+
+| Field Name        | Type    | Description     |
+| :------------- | :----------- | :----------- |
+|  benchmarkTime | TIMESTAMP   | The time at which the benchmark was executed|
+|  fileType  | STRING | The file type tested |
+|  compressionType | STRING  | The method in which the file is compressed  |
+|  numColumns| INTEGER   | The number of columns in each file   |
+|  numRows| INTEGER   | The number of rows in the benchmark table   |
+|  columnTypes | STRING  | The distribution of column types     |
+|  numFiles | INTEGER   | The number of files tested  |
+|  fileSize | INTEGER   | The size of each file   |
+|  stagingDataSize | INTEGER   | The size of the staging table used to generate each file  |
+|  job | RECORD   | -    |'
+|  job.id | STRING   | The id of the load job   |
+|  job.user | STRING   | The user who executed the job   |
+|  job.location | STRING   | The location of the job execution    |
+|  job.startTime | TIMESTAMP   | The time at which the job started    |
+|  job.endTime | TIMESTAMP   | The time at which the job ended   |
+|  job.duration | NUMERIC   | The duration of the job in seconds    |
+|  job.destinationTable | STRING   | The name of the benchmark table  |
+|  job.sourceURI | STRING   | The path of all files loaded   |
+|  job.sourceFormat | STRING   | The format of files loaded    |
```

Comment:
that's a great idea, however I'm that the load jobs in the audit logs don't have data for avgSlots and totalSlotMs for like the query jobs do.

Reply:
`select totalSlotMs, jobId from `data-analytics-pocs.audit.bigquery_audit_log` where eventName ='load_job_complete' and (avgSlots is not null or totalSlotMs is not null)` comes up with no results

Reply:
Actually they do have slots data for load jobs. 
You're getting no results because of your `eventName` filter. s/`load_job_complete`/`load_job_completed`/

Reply:
@annarudy fyi ^

Reply:
ohhh wow thanks for catching that!

Reply:
I ended up using data-analytics-pocs:billing.cloudaudit_googleapis_com_activity_*
 to get totalSlotMs and avgSlots. It looks like data-analytics-pocs.audit.bigquery_audit_log was populated using the query you had sent me (https://github.com/GoogleCloudPlatform/professional-services/blob/master/examples/bigquery-audit-log/bigquery_audit_log.sql#L150), which requires setting up a scheduled query to run. Since scheduled queries have a 15 min limit, the data we need wouldn't be populated in data-analytics-pocs.audit.bigquery_audit_log  quickly enough even if we turned the scheduled query back on.